client/python/samples/resnet_grpc_predict.py

```diff
@@ -0,0 +1,119 @@
+                    help='path to image')
+parser.add_argument('--labels_numpy_path', required=False,
+                    help='numpy in shape [n,1] - can be used to check model accuracy')
+parser.add_argument('--grpc_address', required=False, default='localhost',
+                    help='Specify url to grpc service. default:localhost')
+parser.add_argument('--grpc_port', required=False, default=9000,
+                    help='Specify port to grpc service. default: 9000')
+parser.add_argument('--model_name', default='resnet', help='Model name to query. default: resnet',
+                    dest='model_name')
+parser.add_argument('--model_version', default=0, type=int,
+                    help='Model version to query. Lists all versions if omitted',
+                    dest='model_version')
+parser.add_argument('--input_name', required=False, default='0',
+                    help='Specify input tensor name. default: 0')
+parser.add_argument('--output_name', required=False, default='1463',
+                    help='Specify output name. default: 1463')
+parser.add_argument('--batchsize', default=1, type=int,
+                    help='Number of images in a single request. default: 1',
+                    dest='batchsize')
+args = vars(parser.parse_args())
+
+# configuration
+images_numpy_path = args.get('images_numpy_path')
+labels_numpy_path = args.get('labels_numpy_path')
+address = args.get('grpc_address')
+port = args.get('grpc_port')
+model_name = args.get('model_name')
+model_version = args.get('model_version')
+input_name = args.get('input_name')
+output_name = args.get('output_name')
+batch_size = args.get('batchsize')
+
+
+# images pre-processing
+# returns procssed images and potentially processed labels
+def pre_processing():
+    imgs = np.load(args['images_numpy_path'], mmap_mode='r', allow_pickle=False)
+    imgs = imgs - np.min(imgs)  # Normalization 0-255
+    imgs = imgs / np.ptp(imgs) * 255  # Normalization 0-255
+
+    lbs = None
+    if labels_numpy_path is not None:
+        lbs = np.load(labels_numpy_path, mmap_mode='r', allow_pickle=False)
+
+    while batch_size >= imgs.shape[0]:
+        imgs = np.append(imgs, imgs, axis=0)
+        if args.get('labels_numpy_path') is not None:
+            lbs = np.append(lbs, lbs, axis=0)
+    return imgs, lbs
+
+
+def post_processing(output):
+    for i in range(output.shape[0]):
+        single_result = output[[i], ...]
+        offset = 0
+        if output.shape[1] == 1001:
+            offset = 1
+        ma = np.argmax(single_result) - offset
+        mark_message = ""
+        if labels_numpy_path is not None:
+            if int(lb[i]) == ma:
+                mark_message = "; Correct match."
+            else:
+                mark_message = "; Incorrect match. Should be {} {}".format(lb[i], classes.imagenet_classes[int(lb[i])])
+        print(classes.imagenet_classes[ma], ma, mark_message)
```

Comment:
This could be excluded to another module to make this script more about using client library and less about pre/post processing of the data.